File: mobile-app/lib/models/booking.dart
import 'course.dart';
import 'player.dart';

class Booking {
  final int id;
  final Player player;
  final Course course;
  final String bookingDate;
  final String teeTime;
  final int numberOfPlayers;
  final String status;

  const Booking({
    required this.id,
    required this.player,
    required this.course,
    required this.bookingDate,
    required this.teeTime,
    required this.numberOfPlayers,
    required this.status,
  });

  factory Booking.fromJson(Map<String, dynamic> json) => Booking(
        id: json['id'] as int,
        player: Player.fromJson(json['player'] as Map<String, dynamic>),
        course: Course.fromJson(json['course'] as Map<String, dynamic>),
        bookingDate: json['bookingDate'] as String,
        teeTime: json['teeTime'] as String,
        numberOfPlayers: json['numberOfPlayers'] as int,
        status: json['status'] as String,
      );
}

File: backend/src/main/java/com/golf/model/Booking.java
package com.golf.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalTime;
import java.time.LocalDateTime;

@Entity
@Table(name = "bookings")
@Data
public class Booking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "player_id", nullable = false)
    private Player player;

    @ManyToOne(optional = false)
    @JoinColumn(name = "course_id", nullable = false)
    private Course course;

    @Column(nullable = false)
    private LocalDate bookingDate;

    @Column(nullable = false)
    private LocalTime teeTime;

    @Min(1) @Max(4)
    @Column(nullable = false)
    private Integer numberOfPlayers = 1;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private BookingStatus status = BookingStatus.CONFIRMED;

    private String notes;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    public enum BookingStatus {
        CONFIRMED, CANCELLED, COMPLETED
    }
}
